tools/plugins/com.liferay.ide.sdk.core/src/com/liferay/ide/sdk/core/SDK.java

```diff
@@ -1003,10 +1011,56 @@ public String toXmlString()
         return builder.toString();
     }
 
+    public IStatus reloadProperties() throws CoreException
+    {
+        Map<String,Object> sdkProperties = null;
+        try
+        {
+            sdkProperties = getBuildProperties( true );
+        }
+        catch( Exception e)
+        {
+            throw new CoreException( SDKCorePlugin.createErrorStatus(e.getMessage()));
+        }
+
+        final IStatus sdkValid = validate(sdkProperties);
+
+        if ( sdkValid.isOK() )
+        {
+            BUILD_PROPERTIES = sdkProperties;
+        }
+
+        return sdkValid;
+    }
+
     public IStatus validate()
     {
         MultiStatus status = new MultiStatus( SDKCorePlugin.PLUGIN_ID, IStatus.OK, "", null );
 
+        Map<String,Object> sdkProperties = null;
+        try
+        {
+            sdkProperties = getBuildProperties( false );
+        }
+        catch( Exception e)
+        {
+            status.add( SDKCorePlugin.createErrorStatus( e.getMessage() ) );
+        }
+
+        final IStatus sdkValid = validate(sdkProperties);
+
+        if ( sdkValid.isOK() && BUILD_PROPERTIES == null )
+        {
+            BUILD_PROPERTIES = sdkProperties;
+        }
+
+        return sdkValid;
+    }
+
+    public IStatus validate(final Map<String,Object> sdkProperties)
```

Comment:
We need to cache even the bad ones.  Because if the user has incorrect properties... we still need to cache them or if the user has bad values then we are right back to back JSP performance.